pygmt/datatypes/header.py

```diff
@@ -0,0 +1,278 @@
+                newdims[dim] = "lat"
+
+            # Axis attributes are "X"/"Y"/"Z"/"T" for horizontal/vertical/time axis.
+            attrs[dim]["axis"] = dim.upper()
+            idx = 2 if dim == "y" else 0
+            attrs[dim]["actual_range"] = np.array(self.wesn[idx : idx + 2])
+
+        # Save the lists of dimension names and attributes in the _nc attribute.
+        self._nc = {
+            "dims": [newdims[dim] for dim in dims],
+            "attrs": [attrs[dim] for dim in dims],
+        }
+
+    def get_name(self) -> str:
+        """
+        Get the name of the grid from the grid header.
+
+        Returns
+        -------
+        name
+            The name of the grid.
+        """
+        return "z"
+
+    def get_data_attrs(self) -> dict:
+        """
+        Get the attributes for the data variable from the grid header.
+
+        Returns
+        -------
+        attrs
+            The attributes for the data variable.
+        """
+        attrs: dict[str, Any] = {}
+        attrs["Conventions"] = "CF-1.7"
+        attrs["title"] = self.title
+        attrs["history"] = self.command
+        attrs["description"] = self.remark
+        long_name, units = _parse_nameunits(self.z_units.decode())
+        if long_name:
+            attrs["long_name"] = long_name
+        if units:
+            attrs["units"] = units
+        attrs["actual_range"] = np.array([self.z_min, self.z_max])
+        return attrs
+
+    def get_dims(self):
+        """
+        Get the dimension names from the grid header.
+
+        Returns
+        -------
+        dims : tuple
+            The dimension names.
+        """
+        if not hasattr(self, "_nc"):
+            self._parse_dimensions()
+        return self._nc["dims"]
+
+    def get_dim_attrs(self) -> list:
+        """
+        Get the attributes for each dimension from the grid header.
+
+        Returns
+        -------
+        attrs
+            List of attributes for each dimension.
+        """
+        if not hasattr(self, "_nc"):
+            self._parse_dimensions()
+        return self._nc["attrs"]
+
+    def get_gtype(self) -> int:
+        """
+        Get the grid type from the grid header.
+
+        The grid is assumed to be Cartesian by default. If the x/y dimensions are named
+        "lon"/"lat" or have units "degrees_east"/"degrees_north", then the grid is
+        assumed to be geographic.
+
+        Returns
+        -------
+        gtype
+            The grid type. 0 for Cartesian grid and 1 for geographic grid.
+        """
+        dims = self.get_dims()
+        gtype = 1 if dims[0] == "lat" and dims[1] == "lon" else 0
+        return gtype
+
+    def get_registration(self) -> int:
+        """
+        Get the grid registration from the grid header.
+
+        Returns
+        -------
+        registration
+            The grid registration. 0 for gridline and 1 for pixel.
+        """
+        return self.registration
```

Comment:
We could just remove this and call `self.registration` no?

Reply:
Removed in 79f4c879e1cc0af2ce3f8cbd9d5eaac6a5644ad6.